src/common/sum_tree.jl

```diff
@@ -131,6 +131,24 @@ function Base.empty!(t::SumTree)
     t
 end
 
+function correct_priority(t::SumTree, leaf_ind)
+    p = t.tree[leaf_ind]
+    # walk backwards until p != 0 or until leftmost leaf reached
+    tmp_ind = leaf_ind
+    while p == 0f0 && (tmp_ind-1)*2 > length(t.tree)
```

Comment:
Do we really want to hardcode `Float32(0)` elements for zero or can we get away with `0` and let Julia do the type inference?

Reply:
25ee56c